azure-pipelines.yml

```diff
@@ -252,8 +312,14 @@ jobs:
         ISPARENT: True
   steps:
   - bash: |
+      uuid="$(tr -dc '[:alnum:]' </dev/urandom| head -c 16)"
```

Comment:
there might be simpler : ${BUILD_BUILDNUMBER}

https://docs.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml

Reply:
I've been burned by using build numbers before... probably not so risky here given that the build number is unlikely to ever get reused given the size of the project and number of builds actually occuring.

But maybe something like `Build.BuildId` which is unique for every attempt at a job? (ie. I think if you reran the exact build again, the `Build.BuildNumber` would be the same, but the `Build.BuildId` would be different.)

Obviously the real goal here is to just make sure that whatever the token is, it needs to not exist on docker hub... but I'm just personally always wary of using Build numbers 😕